### ISSUE TITLE
jshint checkstyle output produces invalid "column 0" errors

### ISSUE THREAD

**Issue by purcell** [flycheck/flycheck#578]
Running the latest jshint from npm (v2.1.10) I see warnings with the column number zero:

```xml
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="4.3">
	<file name="app.js">
		<error line="38" column="53" severity="error" message="Missing semicolon." source="jshint.W033" />
		<error line="38" column="54" severity="error" message="Missing semicolon." source="jshint.W033" />
		<error line="38" column="0" severity="warning" message="Implied global &apos;console&apos;" source="jshint.implied-globals" />
		<error line="16" column="0" severity="warning" message="Unused variable: &apos;ctrl&apos;" source="jshint.implied-unuseds" />
	</file>
</checkstyle>
```

This leads to errors getting highlighted on the prior line (37 in this case).

I'm *guessing* this is a problem upstream, since the columns "53" and "54" above lead to flycheck highlighting what I would consider the logical buffer locations -- just thought I'd post it here first, in case you have any insights.

**Comment by lunaryorn** [comment]
@purcell Can you show the corresponding source code?  I wonder whether the columns from jshint are based on zero or one…

**Comment by purcell** [comment]
Ah. I think they might actually be zero-based:

```js
//345678901234567890
var foo = 1

console.log("hello");
```

which flymake marks up like this:

![screen shot 2015-01-27 at 19 23 23](https://cloud.githubusercontent.com/assets/5636/5925243/086e86bc-a65a-11e4-8c18-2935fb42d3d6.PNG)

That note about the missing semicolon looks like it should be on the newline, not the `1`...

**Comment by lunaryorn** [comment]
Can you try to add `flycheck-increment-error-columns` to the `:error-filter` of the jshint checker?  I.e.

```cl
:error-filter (lambda (errors) (flycheck-dequalify-error-ids (flycheck-increment-error-columns errors)))
```

**Comment by purcell** [comment]
That gives this result:

![screen shot 2015-01-27 at 19 34 19](https://cloud.githubusercontent.com/assets/5636/5925454/8af90552-a65b-11e4-9a1e-128594808baf.PNG)

So it fixes the 2 misplaced warnings, but I'm thinking that column 12 looks more "correct" for the "missing semicolon" warning than column 11. I should try a couple more example errors to see what it does...

**Comment by purcell** [comment]
In the following case, I'd expect that the `Unmatched '{'` message would be located on the opening `{`. And column 13 would make more sense for the unexpected `else` error:

![screen shot 2015-01-27 at 19 39 24](https://cloud.githubusercontent.com/assets/5636/5925559/3f829920-a65c-11e4-8513-5c52f3859ff3.PNG)

So I'm thinking that the jshint numbering is inconsistent, and that (perhaps) errors for which they do not have a specific column are being reported with column 0 instead of 1.

**Comment by lunaryorn** [comment]
@purcell I'll take a look tomorrow evening.  Probably we just need to substitute zero columns…

**Comment by purcell** [comment]
Cool, thanks.

**Comment by lunaryorn** [comment]
@purcell What's your jshint version?  My error messages look different from yours…

```
    2   5 warning  W098   'foo' is defined but never used.... (javascript-jshint)
    2  12 warning  W033   Missing semicolon.... (javascript-jshint)
```

**Comment by purcell** [comment]
```
% jshint --version
jshint v2.1.10
```

From npm. Appears to be the latest available version.

**Comment by lunaryorn** [comment]
Not what I have:

```
$ jshint --version
jshint v2.6.0
```

Seems to be the [latest from npm.js](https://www.npmjs.com/package/jshint):

![bildschirmfoto 2015-01-29 um 20 58 37](https://cloud.githubusercontent.com/assets/224922/5965090/a21ea9a4-a7f9-11e4-8095-4760ba8a8845.png)

**Comment by lunaryorn** [comment]
@purcell Did you try with 2.6?

**Comment by purcell** [comment]
Yeah, for some silly `node`-related reason I seemed to be stuck with that older version. I can't reproduce the problem with the newer version, so I'll go ahead and close this. Sorry!
